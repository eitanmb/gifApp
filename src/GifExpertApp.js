import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

//Functional Component FC
const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Legend of Zelda'])
    
    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr />
            <ol>
                {
                    categories.map( category => 
                        ( 
                            <GifGrid 
                                key={category}
                                category={category} 
                            /> 
                        )
                    )
                
                }
            </ol>
        </>

    )

}

export default GifExpertApp;