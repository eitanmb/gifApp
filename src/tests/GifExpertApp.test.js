import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import GifExpertApp from '../GifExpertApp';

describe('Pruebas en GifExpertApp', () => {

    test('debe dibujar el componente correctamente ', () => {

        const wrapper = shallow(<GifExpertApp />);

        expect( wrapper ).toMatchSnapshot();
        
    });

    test('debe mostrar una lista de categories', () => {
        
        const categories = ['Zelda', 'Mario'];

        const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length +1 );

    });
    
    
    
});
