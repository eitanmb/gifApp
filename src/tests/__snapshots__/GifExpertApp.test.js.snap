// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp debe dibujar el componente correctamente  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Legend of Zelda"
      key="Legend of Zelda"
    />
  </ol>
</Fragment>
`;
